### ISSUE TITLE
Hard time with blocking Pandora ads

### ISSUE THREAD

**Issue by tu5** [AdAway/AdAway#901]

**Comment by SMed79** [comment]
Adguard, Adblock Plus, AdBlock or uBlock origin has more advanced filtering options then only DNS in Adaway.

Try **Adblock Browser** https://adblockbrowser.org/

**Comment by Techman-** [comment]
@smed79 Why spam that here? We already know that the folks behind AdBlock Plus are a scum company, turning into an ad firewall by default instead of blocking them.

On mobile, the Android app is what Pandora pretty much requires you to use. Pandora doesn't have a mobile web app (and why would they?) the last time that I checked.

**Comment by SMed79** [comment]
> Why spam that here?

Trying to help instead of trolling.

> turning into an ad firewall by default instead of blocking them

Do you have an example?

**Comment by Techman-** [comment]
@SMed79 We all already know that [the company behind AdBlock Plus charges companies to whitelist their ads so that they can get through by default](https://arstechnica.com/business/2015/02/over-300-businesses-now-whitelisted-on-adblock-plus-10-pay-to-play/). They're basically reverse-profiting from advertisements. That's not how the art of content blockers should work. This is the precise reason why I use uBlock Origin now.

**Comment by SMed79** [comment]
Huh, should i trust ABP or the press troll? Acceptable ads is not an hidden option and [unchecking](https://adblockplus.org/acceptable-ads#optout) a simple box is not a herculean task.

I am here trying to give some help to whoever asks for it so please If you can't think of anything to say to contribute for helping don't say anything at all. Done!

@tu5 
...an alternative of Adblock Browser is to install [Firefox Browser](https://play.google.com/store/apps/details?id=org.mozilla.firefox) then add [uBO addon](https://addons.mozilla.org/firefox/addon/ublock-origin/).
Try also Adguard for Android https://adguard.com/en/adguard-android/overview.html

If does not work for you, just give a real testable case than we can check.

PS: _For who read this issue, sorry for the **Off Topic** ..._

**Comment by tu5** [comment]
@SMed79 Mobile Pandora is an app, not website so any browser will be useless. All I need is proper filters to enter in AdAway.

**Comment by SMed79** [comment]
> Mobile Pandora is an app

Excuse my ignorance on this.

Please link me to this app on the play store and post a screenshot of the unblocked ads.

**Comment by tu5** [comment]
https://play.google.com/store/apps/details?id=com.pandora.android


![screenshot_20170528-105557](https://cloud.githubusercontent.com/assets/18059922/26527435/8b038c88-4394-11e7-9976-14efe97f7e39.png)
![screenshot_20170528-105634](https://cloud.githubusercontent.com/assets/18059922/26527436/8e8f5738-4394-11e7-9702-84015dd15014.png)

**Comment by SMed79** [comment]
No available on my country, give more time to test later behind a VPN.

![Pandora](http://i.imgur.com/oyqAjsR.png)

**Comment by tu5** [comment]
Thanks, this may be helpful. https://apps.evozi.com/apk-downloader/

**Comment by SMed79** [comment]
Tested behind an UK VPN and Pandora app still blocking me :-/

Please try to block this host `admarvel.s3.amazonaws.com` and give feedback.

**Comment by tu5** [comment]
Yeah because Pandora is only available in US, Canada, and Australia. Will try and let you know.

**Comment by tu5** [comment]
No result.

**Comment by SMed79** [comment]
Try this

1. kill pandora app and clean it's cache.
2. follow the indicated in the wiki https://github.com/AdAway/AdAway/wiki/LogDNS
3. post here screenshot of hostnames you see in the log.

**Comment by tu5** [comment]
![screenshot_20170528-143345](https://cloud.githubusercontent.com/assets/18059922/26528783/d74cf3cc-43b2-11e7-8a84-53d34e651cbb.png)
![screenshot_20170528-143351](https://cloud.githubusercontent.com/assets/18059922/26528784/db94570e-43b2-11e7-9e60-6da61ff404f0.png)

**Comment by SMed79** [comment]
The only suspicious host is `gap.sp1.convertro.com` block it and test. 

If the ads is served from pandora.com or his cdn It will be impossible to block whith the hosts file without breaking at the same time the app :-/
